File: 0205-isomorphic-strings/0205-isomorphic-strings.java
import java.util.*;

class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length())
            return false;
        char sList[] = s.toCharArray();
        char tList[] = t.toCharArray();
        HashMap<Character, Character> dict = new HashMap<Character, Character>();
        for(int i = 0; i < s.length(); i++) {
            if(dict.get(sList[i]) == null) {
                if(!dict.values().contains(tList[i]))
                    dict.put(sList[i], tList[i]);
                else
                    return false;
            }
            else if(dict.get(sList[i]) != tList[i])
                return false;
        }
        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        # Hashmap that tracks the mapped characters from one string to another
        sMap = {}
        tMap = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if not sMap.get(s[i]):
                sMap[s[i]] = t[i]
            if not tMap.get(t[i]):
                tMap[t[i]] = s[i]
            if sMap[s[i]] != t[i] or tMap[t[i]] != s[i]:
                return False
        return True